### PR TITLE
Adding process and template for license exceptions

### PR DESCRIPTION
**Pull request by amye**
We're formalizing the process to ask for exceptions and how they get resolved. Comments welcomed.

### CODE REVIEW

**Comment by dims** [comment]
@amye @caniszczyk typically projects will ask for a set of exceptions. Say someone on project wants to cleanup their go.mod/go.sum they will need to file individual issues and debate on each of those.

- Could we please optimize for that case?
- Could we let projects open PRs with the set of projects+licenses and add more details in the PR body?

thanks!

**Comment by amye** [comment]
See https://github.com/cncf/foundation/issues/297 as an example of your first 'set'. 
I don't have an opinion on PRs vs Issues, right now the Issue queue is set up to track.

File: .github/ISSUE_TEMPLATE/license-exception.md [hunk 1]
```diff
@@ -7,9 +7,12 @@ assignees: caniszczyk, amye
...
-- [ ] What is the alternative license?
-- [ ] If this is newly created code, why can't it be under Apache-2.0? 
-- [ ] Is this an existing 3rd party open source project?
-- [ ] How does this code integrate with or interact with, if at all, other components of CNCF? 
-- [ ] How will the code be maintained? Who is responsible? 
-- [ ] How will the code be kept up to date with security patches?
+- [ ] What is the third party component (name, URL, and version if available)?
+- [ ] What license(s) is it provided under?
+- [ ] If it is Apache-2.0 for code or CC-BY-4.0 for documentation, no further exception request is required.
+- [ ] Does it meet the Allowlist policy? If so, no further exception request is required.
```
Comment (dims) [inline .github/ISSUE_TEMPLATE/license-exception.md:13]: let's add a link to the policy? https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md

**Comment by dims** [comment]
One more question i think would make sense:
- Is this used for just building of artifacts for distribution? 
(inspired by https://www.apache.org/legal/resolved.html#build-tools)

**Comment by amye** [comment]
Oh interesting, let me think about how I might word that so that the question's clear. Still working through this!

**Comment by amye** [comment]
@krook to review, I think yaml makes more sense as a template. 
Right now, we prioritizing projects actively needing exceptions over 'generally in use'.

**Review by TheFoxAtWork — COMMENTED** [review]
so suggested tweaks

File: .github/ISSUE_TEMPLATE/license-exception.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+title: "[License Exception Request] [Project] [License]"
+labels: licensing
+assignees: krook, amye
+---
+
+For all license exception requests, we need the following information in order to help make a decision to grant an exception.
+
+- [ ] What is the third party component (name, URL, and version if available)?
+- [ ] What license(s) is it provided under?
+- [ ] If it is Apache-2.0 for code or CC-BY-4.0 for documentation, no further exception request is required.
```
[the remark this answers is not in the document]
Comment (amye) [reply .github/ISSUE_TEMPLATE/license-exception.md]: Replaced with a .yaml based template and yes!

**Comment by krook** [comment]
> @krook to review, I think yaml makes more sense as a template. Right now, we prioritizing projects actively needing exceptions over 'generally in use'.

@amye I committed onto your branch a modified version of the YAML issue template. My commit is signed off, but clearly DCO is confused with multiple authors on the branch. I can either force it to pass, or rebase and sign all the previous commits in the branch (yours as mine). Do you have a preference?

@amye @TheFoxAtWork Here is a look at the [GitHub issue template in action in another test repo](https://github.com/krook/test-issues/issues/new/choose), as it can't be previewed in this repo outside the main branch.

**Comment by amye** [comment]
Resolved the DCO issue, likely my error caused it.

File: allowed-third-party-license-policy.md [hunk 3]
```diff
@@ -57,3 +57,19 @@ List.
 [Allowlist License Policy]: #cncf-allowlist-policy
 [Approved Licenses]: #approved-licenses-for-allowlist
 [CNCF Charter]: https://github.com/cncf/foundation/blob/master/charter.md
+
+### Process for applying for an exception
+
+This outlines the process for applying for an exemption to the Intellectual Property policy referenced above.
+
+- A CNCF project maintainer can file a request for an exception using the issue template.
```
[the remark this answers is not in the document]
Comment (amye) [reply allowed-third-party-license-policy.md:65]: That's in this PR so it won't resolve yet

**Review by TheFoxAtWork — APPROVED** [review]

**Review by krook — APPROVED** [review]

**Comment by krook** [comment]
@amye is this good to merge?

**Comment by amye** [comment]
> @amye is this good to merge?
I think so! Happy to scream-test and see where improvements need to happen.

**Pull request merged**